--- get.Mumps	2016-11-25 03:59:59.000000000 -0800
+++ get.Mumps	2019-11-09 17:39:23.000000000 -0800
@@ -28,7 +28,7 @@
 rm -f MUMPS*.tgz
 
 echo "Downloading the source code from ..."
-$wgetcmd http://mumps.enseeiht.fr/MUMPS_${mumps_ver}.tar.gz
+$wgetcmd https://sourceforge.net/projects/myosin/files/ipopt/MUMPS_${mumps_ver}.tar.gz
 
 echo "Uncompressing the tarball..."
 gunzip -f MUMPS_${mumps_ver}.tar.gz
